src/Simulation/Simulators/QuantumSimulator/ApplyControlledX.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.Simulators
+{
+    public partial class QuantumSimulator
+    {
+        public virtual Func<(Qubit, Qubit), QVoid> ApplyControlledX_Body() => (args) =>
```

Comment:
Thinking about how the generated code needs to look like, since this is what would be implemented manually, we should expose it with a more natural signature and push the "plumbing" into the generated piece.
```
public virtual void ApplyControlledX_Body(Qubit control, Qubit target) { ... }
```

Reply:
I agree, that is much nicer! I'm going to rework the interfaces to use this approach, but first I'd like to get the e2e build working to ensure I have a clean foundation for further changes.

Reply:
Ok, I've adjusted the interface style to match this suggestion. Now I'll work on changing the C# generation so that we don't need to use the type parameters for overriding.

Reply:
Thanks @ScottCarda-MS for making the changes needed to support concreate class generation for intrinsics! It makes all of this cleaner.